File: errata-rails/examples/ruby/qpid/simple.rb
#!/usr/bin/ruby
require 'qpid_handler'

qpid = QpidHandler.new('qpid.engineering.redhat.com')

qpid.topic_subscribe('eso.topic', 'errata.#') do |content, message|
  puts \
    '-'*28,
    Time.now.utc,
    "Headers: #{message.getProperties.inspect}",
    "Subject: #{message.getSubject.inspect}",
    "Content: #{content.inspect}"
end

qpid.listen

File: errata-rails/examples/python/qpid/simple.py
#!/usr/bin/python

from errata_listener import ErrataListener
from datetime import datetime

class SimpleListener(ErrataListener):
    def message_handler(self, msg):
        self.say('--------------------------')
        self.say('Received: ' + str(datetime.now()))
        self.say('Subject: ' + str(msg.subject))
        self.say('Content: ' + str(msg.content))

if __name__ == '__main__':
    SimpleListener().listen()
